### PR TITLE
Add backend support for organizing flows into folders

### PR DESCRIPTION
**Pull request by robluton**
## What's Changed

- Added `type` column to `directus_folders` to distinguish file-library folders (`'assets'`) from flow folders (`'flows'`)
- Added `folder` column to `directus_flows` that references `directus_folders`
- Added migration that backfills existing folders to `'assets'` and leaves flows untouched
- Registered the new fields and relation in system-data
- Updated shared types for the folder and flow changes

## Tested Scenarios

- Migration runs cleanly on existing installs
- File library returns only asset folders
- Deleting a folder nulls the flow's `folder` without deleting the flow

## Review Notes / Questions / Concerns

- `type` defaults to `'assets'`; flow-folder consumers must scope to `type = 'flows'`
- Rolling back the migration does not remove any folders that may have been using the type 'flows', but does remove the type column. Running the migration again sets the default folder type to 'assets.' This means any folders previously created for flows will now show up in the assets library. I'm considering this a rare edge case, but open to suggestions if we think this is an issue we should handle differently. 

## Checklist

> Leave unchecked where not applicable

- [x] Tests added/updated
- [ ] Documentation PR created in [directus/docs](https://github.com/directus/docs) 
- [x] OpenAPI updated
  - [x] Local specs package (`@directus/specs`)
  - [ ] PR created in [directus/openapi](https://github.com/directus/openapi) 
- [x] SDK (`@directus/sdk`) updated to reflect the changes
- [x] Types (`@directus/types`) updated to reflect the changes
- [ ] GraphQL schema updated to reflect the changes
- [x] System data (`@directus/system-data`) updated for changes to system collections/fields/relations
- [x] Database migration added for schema/system changes
- [ ] Environment variables documented for new/changed config
- [ ] App translations added for new user-facing strings
- [ ] Security implications apply

Fixes [CMS-2934](https://linear.app/directus/issue/CMS-2934/build-out-flow-folder-interface)

### CODE REVIEW

**Comment by kheiner** [comment]
Don't forget to update SDK Schemas.

File: .changeset/wild-dryers-wear.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+'@directus/api': minor
+'@directus/system-data': patch
+'@directus/types': patch
+'@directus/cli': patch
+'@directus/sdk': minor
+'@directus/specs': patch
+---
+
+Add folder support to flows, allowing flows to be organized into folders
```
Comment (AlexGaillard) [inline .changeset/wild-dryers-wear.md]:
```suggestion
Added folder support to flows, allowing flows to be organized into folders
```

Comment (AlexGaillard) [inline .changeset/wild-dryers-wear.md]:
```suggestion
Added folder support to flows, allowing flows to be organized into folders
```

File: api/src/database/migrations/20260818A-add-flow-folders.ts
Comment (AlexGaillard) [inline api/src/database/migrations/20260818A-add-flow-folders.ts:1]: So I foresee the following problem here once we have the frontend doing its thing, but I don't really know if there's a way around it:

- User migrates up -> all existing folders get type "assets"
- User adds a folder(s) for flows with type "flows"
- User migrates down -> type column removed from all folders
- User migrates up -> all existing folders get type "assets", even those previously for flows

Comment (robluton) [reply api/src/database/migrations/20260818A-add-flow-folders.ts:1]: Yes this is what I called out in the PR description:

> Rolling back the migration does not remove any folders that may have been using the type 'flows', but does remove the type column. Running the migration again sets the default folder type to 'assets.' This means any folders previously created for flows will now show up in the assets library. I'm considering this a rare edge case, but open to suggestions if we think this is an issue we should handle differently.

Comment (kheiner) [reply api/src/database/migrations/20260818A-add-flow-folders.ts:1]: Could it be done the way `directus_collections.group` handles grouping?

Comment (ComfortablyCoding) [reply api/src/database/migrations/20260818A-add-flow-folders.ts:1]: Collections uses its own paradigm for indicating what constitutes a folder (`schema:null`), so we’d need to establish a convention for flows that conveys “folder” (e.g., `trigger: null`). That’s not ideal, since it isn’t particularly intuitive.

Comment (kheiner) [reply api/src/database/migrations/20260818A-add-flow-folders.ts:1]: > ...’d need to establish a convention for flows that conveys “folder” (e.g., trigger: null). That’s not ideal, since it isn’t particularly intuitive.

I agree with that; I'm thinking about my expectations if I were to use the API `GET /folders` it would show the folders for assets (intuitive, like a directory) and folders for flows (the flows don't/won't contain anything?). You also are introducing the need to always filter this call for every path that users folders (today it's `api/src/ai/tools/*` but who knows where it goes in maintaining future features). It feels like there's trade-offs either way. Another idea for a solution is a system-only table for `directus_flow_folders`.

I don't think it's going to be very common for an upgrade/downgrade so the concern is narrow but maintainability for the new pattern of `folder.type` seems like a lot to carry (especially since the user expectation of a folder is likely to be that it holds stuff, not just a group-label).

Comment (robluton) [reply api/src/database/migrations/20260818A-add-flow-folders.ts:1]: @kheiner these are good points. Our original plan was to have a dedicated system collection called `directus_flows_folders`. There are trade-offs to each approach, but we feel like adapting the existing directus_folders to handle other types will be easier to manage than having separate endpoints, etc... for a new collection. 

Organizing other areas (not just flows) using folders is something we want to explore, and having a shared model for this feels like the right choice. 

But I do agree that always having to filter every path using folders is a significant drawback to this approach, but acceptable given the alternatives.

Comment (kheiner) [reply api/src/database/migrations/20260818A-add-flow-folders.ts:1]: Thanks for clarifying; If it becomes an established pattern I'm sure it will be easier to understand and won't feel as foreign to have 'folders-as-labels'.

I'm fine either way (not that I believe I have any control) I just wanted to surface an outside perspective.

I totally get that it's hard to build in public and I don't want to slow/stop Directus from being great -- thanks for entertaining my concerns and showing that you're thinking through these things deeply.

**Review by AlexGaillard — APPROVED** [review]
:cool::beans:
